### PR TITLE
Add watchdogutil to control the hw watchdog

### PR DESCRIPTION
**Pull request by sujinmkang**
**- What I did**
Add a command line utility to control the hw watchdog arm/disarm
**- How I did it**
Add watchdogutil
**- How to verify it**
Check the syslog 
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
```
user@server:~$ sudo watchdogutil status
Status : Unarmed
user@server:~$ sudo watchdogutil arm
Watchdog armed for 180 seconds
user@server:~$ sudo watchdogutil disarm
Watchdog disarmed successfully
```

### CODE REVIEW

File: watchdogutil/main.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    return 0
+
+
+# ==================== CLI commands and groups ====================
+
+
+# This is our main entrypoint - the main 'watchdogutil' command
+@click.group()
+def cli():
```
Comment (jleveque) [inline watchdogutil/main.py:84]: Suggest to name the main group the same name as the utility, mainly for readability (i.e., `def watchdogutil`). This way, it's easier to understand the hierarchy when reading the code. You will also need to change all references in this file as well as change setup.py to read `watchdogutil = watchdogutil.main:watchdogutil`. I want to update all existing utilities which currently use "cli" in the future, also.

File: watchdogutil/main.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        sys.exit(2)
+
+# 'version' subcommand
+@watchdogutil.command()
+def version():
+    """Display version info"""
+    click.echo("watchdogutil version {0}".format(VERSION))
+
+# 'is_armed' subcommand
+@watchdogutil.command()
```
Comment (jleveque) [inline watchdogutil/main.py:103]: As of Click 7.0, underscores are replaced with hyphens, thus this subcommand will be `is-armed`. Please change comment to match, and also to ensure backward-compatibility when using older versions of Click, force the name by passing the name as a parameter to command(). I.e.,

```
# 'is-armed' subcommand
@watchdogutil.command('is-armed')
```
The same applies to the `get_remaining_time` subcommand.

Actually, I'm wondering if we should combine the 'is-armed' and 'get-remaining-time` commands into a 'status' command which outputs the armed status and the remaining time, or 'N/A' if the watchdog is not armed.

**Comment by sujinmkang** [comment]
@jleveque   PR description updated

File: watchdogutil/main.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+def version():
+    """Display version info"""
+    click.echo("watchdogutil version {0}".format(VERSION))
+
+# 'status' subcommand
+@watchdogutil.command()
+def status():
+    """Check the watchdog status with remaining_time if it's armed"""
+    click.echo(str(get_watchdog_status()))
```
Comment (jleveque) [inline watchdogutil/main.py:110]: This output should be in a human-readable form. Maybe use the tabulate module and add a header? Something like,

```
Status    Time Remaining
--------  ----------------
Armed     2 seconds
```

```
Status    Time Remaining
--------  ----------------
Unarmed   N/A
```

File: watchdogutil/main.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+# 'disarm' subcommand
+@watchdogutil.command()
+def disarm():
+    """Disarm HW watchdog"""
+    click.echo(str(platform_watchdog.disarm()))
```
Comment (jleveque) [inline watchdogutil/main.py:116]: Again, output should be human-readable, like "Watchdog disarmed" or "Failed to disarm watchdog"

File: watchdogutil/main.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+# 'arm' subcommand
+@watchdogutil.command()
+@click.option('-s', '--seconds', default=180, type=int, help="the default timeout of HW watchdog")
+def arm(seconds):
+    """Arm HW watchdog"""
+    click.echo(str(platform_watchdog.arm(seconds)))
```
Comment (jleveque) [inline watchdogutil/main.py:123]: Again, output should be human-readable, like "Watchdog armed for 180 seconds" or Failed to arm watchdog"

File: watchdogutil/main.py (hunk 3 above)
Comment (sujinmkang) [reply watchdogutil/main.py]: We can use tabulate for this output but there is only one entry of two value returns for armed and unarmed status so I'd like to use plane text for this output.

Comment (jleveque) [reply watchdogutil/main.py]: OK. Then how about the following?:

```
Status: Armed
Time remaining: 2 seconds
```

```
Status: Unarmed
```

**Comment by renukamanavalan** [comment]
retest this please

File: watchdogutil/main.py (hunk 3 above)
Comment (sujinmkang) [reply watchdogutil/main.py]: I changed as below but I will change as you suggested. It looks clear.
```
Watchdog armed with remaining 2 seconds
```
```
Watchdog unarmed
```

File: watchdogutil/main.py [hunk 6]
```diff
@@ -106,9 +106,10 @@ def status():
     status = platform_watchdog.is_armed()
     remaining_time = platform_watchdog.get_remaining_time()
     if status is True:
-        click.echo("Watchdog armed with remaining {} seconds".format(remaining_time))
+        click.echo("Status : Armed")
```
Comment (jleveque) [inline watchdogutil/main.py:109]: Remove space before `:`

File: watchdogutil/main.py [hunk 7]
```diff
@@ -106,9 +106,10 @@ def status():
...
+        click.echo("Time remaining : {} seconds".format(remaining_time))
```
Comment (jleveque) [inline watchdogutil/main.py:110]: Remove space before `:`

File: watchdogutil/main.py [hunk 8]
```diff
@@ -106,9 +106,10 @@ def status():
...
     else:
-        click.echo("Watchdog unarmed")
+        click.echo("Status : Unarmed")
```
Comment (jleveque) [inline watchdogutil/main.py:112]: Remove space before `:`

File: watchdogutil/main.py [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+try:
+    import sys
+    import os
+    import click
+    import syslog
+    import sonic_platform
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
+
+VERSION = '2.0'
```
Comment (jleveque) [inline watchdogutil/main.py:17]: Why not 1.0?

File: watchdogutil/main.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+        click.echo("Failed to disarm Watchdog")
+
+# 'arm' subcommand
+@watchdogutil.command()
+@click.option('-s', '--seconds', default=180, type=int, help="the default timeout of HW watchdog")
+def arm(seconds):
+    """Arm HW watchdog"""
+    result = int(platform_watchdog.arm(seconds))
+    if result == seconds:
+        click.echo("Watchdog armed for {} seconds".format(seconds))
```
Comment (jleveque) [inline watchdogutil/main.py:132]: According to the API definition, if the watchdog doesn't support the number of seconds passed to `arm()`, it should set it to the next greater value (see: https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/watchdog_base.py#L24)

Thus, I think this should be changed to read:
```
    if result >= seconds:
        click.echo("Watchdog armed for {} seconds".format(result))
```

File: watchdogutil/main.py [hunk 11]
```diff
@@ -128,10 +128,10 @@ def disarm():
 def arm(seconds):
     """Arm HW watchdog"""
     result = int(platform_watchdog.arm(seconds))
-    if result == seconds:
-        click.echo("Watchdog armed for {} seconds".format(seconds))
-    else:
+    if result < 0:
         click.echo("Failed to arm Watchdog for {} seconds".format(seconds))
+    else:
+        click.echo("Watchdog armed for {} seconds".format(results))
```
Comment (jleveque) [inline watchdogutil/main.py:134]: Typo: s/results/result/

File: watchdogutil/main.py [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+def arm(seconds):
+    """Arm HW watchdog"""
+    result = int(platform_watchdog.arm(seconds))
+    if result < 0:
+        click.echo("Failed to arm Watchdog for {} seconds".format(seconds))
+    else:
+        click.echo("Watchdog armed for {} seconds".format(result))
+
+if __name__ == '__main__':
+    cli()
```
Comment (jleveque) [inline watchdogutil/main.py:137]: `cli()` has been renamed to `watchdogutil()` due to one of my earlier comments. This needs to be updated, also.

File: watchdogutil/main.py [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+try:
+    import sys
+    import os
+    import click
+    import syslog
+    import sonic_platform
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
+
+VERSION = '1.0'
```
Comment (jleveque) [inline watchdogutil/main.py:17]: Suggest to replace single-quotes with double-quotes to align with prevailing style in the file.

File: watchdogutil/main.py (hunk 12 above)
Comment (sujinmkang) [reply watchdogutil/main.py:137]: good catch. I don't know how I missed. :)

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Pull request merged**